cmd/upgrade/executor/resource.go

```diff
@@ -52,28 +53,41 @@ func NewUpgradeResourceJob() *cobra.Command {
 		Long:    resourceUpgradeCmdHelpText,
 		Example: `upgrade resource`,
 		Run: func(cmd *cobra.Command, args []string) {
-			util.CheckErr(options.InitializeFromUpgradeTaskResource(cmd), util.Fatal)
-			util.CheckErr(options.RunPreFlightChecks(cmd), util.Fatal)
-			util.CheckErr(options.RunResourceUpgradeChecks(cmd), util.Fatal)
-			util.CheckErr(options.InitializeDefaults(cmd), util.Fatal)
-			util.CheckErr(options.RunResourceUpgrade(cmd), util.Fatal)
+			upgradeTaskCRName := getUpgradeTaskLabel()
+			openebsNamespace := getOpenEBSNamespace()
+			upgradeTaskList, err := utask.NewKubeClient().
+				WithNamespace(openebsNamespace).
+				List(metav1.ListOptions{
+					LabelSelector: upgradeTaskCRName,
+				})
+			util.CheckErr(err, util.Fatal)
+
+			for _, cr := range upgradeTaskList.Items {
```

Comment:
We should probably add a check here to make sure atleast one resource is found to upgrade. Can be taken up as next PR.

Reply:
Done